src/VisualStudio/CSharp/Test/EditorConfigSettings/DataProvider/DataProviderTests.cs

```diff
@@ -171,7 +171,7 @@ public void TestGettingCodeStyleSettingsProviderLanguageServiceAsync()
             var optionsWithUI = CSharpCodeStyleOptions.EditorConfigOptions
                 .Remove(CSharpCodeStyleOptions.PreferredModifierOrder);
 
-            AssertEx.SetEqual(optionsWithUI, dataSnapShot.Select(setting => setting.Key.Option));
+            AssertEx.SetEqual(optionsWithUI.OrderBy(o => o.Name), dataSnapShot.Select(setting => setting.Key.Option).OrderBy(o => o.Name));
```

Comment:
>SetEqual [](http://example.com/codeflow?start=21&length=8)

Why is the ordering necessary for SetEqual?

Reply:
So that when they differ, the diff you get isn't awful. Basically, the diff function does things linearly, which makes for awful adds/removes/moves :-)